crates/autopilot/src/domain/events/mod.rs

```diff
@@ -0,0 +1,205 @@
+use {
+    crate::{
+        boundary,
+        domain::{self, eth},
+        infra::{
+            self,
+            persistence::{self, transaction::Transaction},
+        },
+    },
+    anyhow::Context,
+    ethrpc::current_block::RangeInclusive,
+    std::sync::Arc,
+};
+
+pub struct Events {
+    persistence: Arc<infra::Persistence>,
+}
+
+/// Error type for the events module.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("persistence error: {0:?}")]
+    Persistence(#[from] persistence::Error),
+    #[error("unexpected event data: {0:?}")]
+    Encoding(anyhow::Error),
+}
+
+pub mod settlement {
+    use super::*;
+
+    /// An order was fully or partially traded.
+    pub struct Trade {
+        pub block_number: u64,
+        /// The index of the event in the block.
+        pub log_index: usize,
```

Comment:
I would like it if we could easily associate a tx hash with each event in the long term.
Otherwise you theoretically always have to keep in mind that a single block might contain multiple settlement transactions.